### ISSUE TITLE
Investigate failing Travis builds

### ISSUE THREAD

**Issue by ecurtin** [SparkTC/spark-bench#56]
Builds that pass on ecurtin/spark-bench are failing when merged to SparkTC/spark-bench.

The same exact commit is [passing here](https://travis-ci.org/ecurtin/spark-bench/builds/270549614) but [failing here](https://travis-ci.org/SparkTC/spark-bench/builds/270543520).

My hypothesis is that it might have to do with differences between Precise and Trusty images since that appears to be the only material difference between the two builds.

**Comment by ecurtin** [comment]
Confirmed. The issue stems from the difference in Linux distributions used for the builds. 
Test: https://github.com/ecurtin/spark-bench/commit/d722bdda6546
Result: https://travis-ci.org/ecurtin/spark-bench/builds/270558986

This is likely due to access permissions on files created by the build.

**Comment by ecurtin** [comment]
I have been investigating the root cause of the failure in this branch: https://github.com/ecurtin/spark-bench/tree/bugfix/travis-builds-failing-in-new-repo

The issue is not permissions. The travis user has full access to all files, and all files are being written in the same place.

The tests that are failing in Trusty are tests in which spark-bench-launch submits spark-bench to spark-submit. All the other self-contained tests are passing. The spark-submit on these failing tests fails with no error and no output, just a return code of 137. StackOverflow seems to suggest this might be the JVM OOM killer at work. https://travis-ci.org/ecurtin/spark-bench/builds/270944411#L7253

I pulled down the Trusty JVM container from TravisCI's DockerHub and tried to reproduce the failure locally. No such luck, all the tests passed.

I am opening a support ticket with TravisCI to get this sorted out. In the meantime, I have submitted a bandaid PR that requires the dist be set to Precise. TravisCI will eventually EOL the Precise containers, so this Trusty issue will need to be addressed sooner or later.
